gcloud/storage/bucket.py

```diff
@@ -423,7 +448,13 @@ def delete_blobs(self, blobs, on_error=None, client=None):
                 blob_name = blob
                 if not isinstance(blob_name, six.string_types):
                     blob_name = blob.name
-                self.delete_blob(blob_name, client=client)
+
+                generation = None
+                if hasattr(blob, 'generation'):
```

Comment:
`generation` is a property so `hasattr` is not needed. You can bunch this in with the `not isinstance(blob_name, six.string_types)` above (since that represents a `Blob` object)